### PR TITLE
Fix a bunch of TODOs

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

- Just fixing some old TODOs
- Remove unused variable in BlobSidecarManager
- Couple of nits regarding comments
- Enabled javadoc task (fixed couple of `malformed html` errors in comments and it seems ok)
- Removed inapplicable Java 8 specific javadoc settings

## Fixed Issue(s)
N/A

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/beaconstate/common/TransitionCaches.java [hunk 1]
```diff
@@ -180,7 +180,7 @@ public Cache<UInt64, BLSPublicKey> getValidatorsPubKeys() {
    * (validator pub key) -> (validator index) cache
    *
    * <p>WARNING: May contain mappings for public keys of validators that are not yet registered in
-   * this state (but when registered are guaranteed to be at that index). Check index < total
+   * this state (but when registered are guaranteed to be at that index). Check index &lt; total
```
Comment (zilm13) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/beaconstate/common/TransitionCaches.java]: Maybe use `{@literal < }`, it's more readable both in code and in javadoc?

File: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/LeafNode.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ public interface LeafNode extends TreeNode, LeafDataNode {
   /** The {@link LeafNode} with empty data */
   LeafNode EMPTY_LEAF = ZERO_LEAVES[0];
 
-  /** Creates a basic Leaf node instance with the data <= 32 bytes */
+  /** Creates a basic Leaf node instance with the data &le; 32 bytes */
```
Comment (zilm13) [inline infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/LeafNode.java]: I'd bet for using `@literal` in such cases

File: beacon/sync/src/testFixtures/java/tech/pegasys/teku/beacon/sync/SyncingNodeManager.java [hunk 3]
```diff
@@ -114,15 +116,51 @@ public static SyncingNodeManager create(
...
+    final FutureItems<SignedBeaconBlock> futureBlocks =
+        FutureItems.create(SignedBeaconBlock::getSlot, mock(SettableLabelledGauge.class), "blocks");
+    final Map<Bytes32, BlockImportResult> invalidBlockRoots = LimitedMap.createSynchronized(500);
+    final Map<Bytes32, InternalValidationResult> invalidBlobSidecarRoots =
+        LimitedMap.createSynchronized(500);
+
+    final FutureItems<SignedBlobSidecar> futureBlobSidecars =
+        FutureItems.create(
+            SignedBlobSidecar::getSlot, mock(SettableLabelledGauge.class), "blob_sidecars");
+    final BlobSidecarValidator blobSidecarValidator =
```
Comment (zilm13) [inline beacon/sync/src/testFixtures/java/tech/pegasys/teku/beacon/sync/SyncingNodeManager.java]: I'm not sure on all these changes. We have Phase0 spec here and not covering any blobSidecars behavior in tests using this. I'd just remove both TODOs and keep everything related to blob sidecars here as NOOP

File: build.gradle [hunk 4]
```diff
@@ -289,14 +288,9 @@ allprojects {
...
 
-  // Normalize Xdoclint behaviour across JDKs (OpenJDK 8 is more lenient than Oracle JDK by default).
   javadoc {
-    options.addStringOption('Xdoclint:all', '-quiet')
-    if (JavaVersion.current() == JavaVersion.VERSION_1_8) {
-      options.addStringOption('Xwerror', '-quiet')
-    } else {
-      options.addStringOption('Xwerror', '-html5')
-    }
+    options.addStringOption('Xdoclint:all,-missing', '-quiet')
```
Comment (zilm13) [inline build.gradle:292]: I've tried javadoc on:
openjdk11 - failed, number of error. As we require minimum Java11 I think we may need to support this, but question is should we support javadoc options outside of our CI? I'm not sure on this
openjdk18 - all good
oracle jdk11 - same errors as in openjdk11

From my feelings recent javadoc version is more flexible on converting `<, >, <=, >=` things.
Overall I tend more to ignore JDK11 errors as  it's a support thing and it works on some recent JDKs which we use on our CI.

File: beacon/sync/src/testFixtures/java/tech/pegasys/teku/beacon/sync/SyncingNodeManager.java (hunk 3 above)
Comment (StefanBratanov) [reply beacon/sync/src/testFixtures/java/tech/pegasys/teku/beacon/sync/SyncingNodeManager.java]: Thanks. You are right. I will remove it!

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/beaconstate/common/TransitionCaches.java (hunk 1 above)
Comment (StefanBratanov) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/beaconstate/common/TransitionCaches.java]: Yeah agree looks better, will change tomorrow

**Review by zilm13 — APPROVED** [review]
LGTM
I'd just add a spaces in literals like `{@literal < }` for more readability

**Pull request merged**